File: 4240-find-the-smallest-balanced-index/find-the-smallest-balanced-index.py
class Solution:
    def smallestBalancedIndex(self, nums: list[int]) -> int:
        leftSum = sum(nums)
        rightPdt = 1

        for i in range(len(nums) - 1, -1, -1):
            leftSum -= nums[i]
            if leftSum == rightPdt:
                return i
            elif leftSum < rightPdt:
                break
            rightPdt *= nums[i]

        return -1

File: 4240-find-the-smallest-balanced-index/find-the-smallest-balanced-index.java
class Solution {
    public int smallestBalancedIndex(int[] nums) {
        long leftSum = 0;
        for (int num : nums) {
            leftSum += num;
        }
        long rightPdt = 1;
        // System.out.println(leftSum);

        // only one balanced index
        for (int i = nums.length - 1; i >= 0; i--) {
            leftSum -= nums[i];
            if (leftSum == rightPdt) {
                return i;
            } else if (leftSum < rightPdt) {
                break;
            }
            rightPdt *= nums[i];
        }

        return -1;
    }
}